docs/docs/services.md

```diff
@@ -725,3 +725,300 @@ const createPost = (input) => {
+
+What if you want to expire a cache regardless of whether the data itself has changed? Maybe you make a UI change where you now show a product's SKU on the page where you didn't before. You weren't previously selecing the `sku` field out of the database, and so it hasn't been cached. But now that you're showing it you'll need to add it the list of fields to return from the service. One solution would be forceably update all of the `updatedAt` fields in the database. But a) Prisma won't easily let you do this since it think it controls that column, and b) every product is going to appear to have been edited at the same time, when in fact nothing changed—you just needed to bust the cache.
+
+An easier solution to this problem would be to add some kind of version number to your cache key that you are in control of and can change whenever you like. Something like appending a `v1` to the key: `v1-product-${id}-${updatedAt}`
+
+And this key is our final form: a unique, but flexible key that allows us to expire the cache on demand (change the version) or automatically expire it when the record itself changes.
+
+#### Expiration-based Keys
+
+You can skirt these issues about what data is changing and what to include or not include in the key by just setting an expiration time on this cache entry. You may decide that if a change is made to a product, it's okay if users don't see the change for, say, an hour. In this case just set the expiration time to 3600 seconds and it will automatically be re-built, whether something changed in the record or not:
+
+```js
+cache(`product-${id}`, () => {
+  // ...
+}, { expires: 3600 })
+```
+
+This leads to your product cache being rebuilt every hour, even though you haven't made any changes that are of consequence to the user. But that may be we worth the tradeoff versus rebuilding the cache when *no* useful data has changed (like the `views` column being updated).
+
+#### Global Cache Key Prefix
+
+Just like the `v1` we added to the `product` cache key above, you can globally prefix a string to *all* of your cache keys:
+
+```js title=api/src/lib/cache.js
+export const { cache, cacheFindMany } = createCache(client, {
+  logger,
+  timeout: 500,
+  // highlight-next-line
+  prefix: 'alpha',
```

Comment:
Would this be a good way to make sure you don't accidentally cache data for another user? Like have a prefix like:

`prefix: context.currentUser?.id || 'public'`

Reply:
This is the global config, which only runs at startup, and so wouldn't have a `context` available! But even if worked I wouldn't recommend it—everything that's cached would be unique to the user, which would be massive overkill and kind of defeat the purpose. Imagine caching another copy of the homepage list of blog posts uniquely for every user, even though every user actually sees the same list!

In the next section I discuss including the `currentUser` in your cache key, so hopefully that gives them a good start at creating a unique key when needed, but not so unique that nothing is ever shared among users.

Reply:
Ah I see... so this code is called once, and _before_ the currentUser is populated in the context. 

I think having a way to cache auth requests would still be beneficial - but now that I think about it, you're right that we can't globally configure it (defeats the whole purpose of the cache in public requests)

Reply:
You can always include the `currentUser` data in the individual key, and there's a whole section in the docs about it!